lib/Monitor/Elembio/RunFolder.pm

```diff
@@ -0,0 +1,556 @@
+package Monitor::Elembio::RunFolder;
+
+use Moose;
+use Carp;
+use Readonly;
+use JSON;
+use File::Basename;
+use File::Spec::Functions qw( catfile catdir );
+use DateTime;
+use List::Util 'sum';
+use DateTime::Format::Strptime;
+use Perl6::Slurp;
+use Try::Tiny;
+use File::Find;
+
+use npg_tracking::Schema;
+use Monitor::Elembio::Enum qw( 
+  $BASECALL_FOLDER
+  $CONSUMABLES
+  $CYCLE_FILE_PATTERN
+  $CYCLES
+  $DATE
+  $FLOWCELL
+  $FOLDER_NAME
+  $INSTRUMENT_NAME
+  $INSTRUMENT_TABLE
+  $LANES
+  $RUN_PARAM_FILE
+  $RUN_STATUS_COMPLETE
+  $RUN_STATUS_INPROGRESS 
+  $RUN_TABLE
+  $RUN_UPLOAD_FILE
+  $RUNLANE_TABLE
+  $SERIAL_NUMBER
+  $SIDE
+  $TIME_PATTERN
+  $USERNAME
+);
+
+with qw[
+  WTSI::DNAP::Utilities::Loggable
+];
+
+our $VERSION = '0';
+
+=head2 runfolder_path
+
+Path string to a run folder.
+
+=cut
+has q{runfolder_path} => (
+  isa           => q{Str},
+  is            => q{ro},
+  required      => 1,
+);
+
+=head2 npg_tracking_schema
+
+Schema object for the tracking database connection.
+
+=cut
+has q{npg_tracking_schema}  => (
+  isa        => 'npg_tracking::Schema',
+  is         => q{ro},
+  required   => 1,
+);
+
+=head2 tracking_run
+
+Record representation of a run in the tracking database.
+
+An Elembio run is defined by the tuple flowcell_id, folder_name,
+id_instrument.
+The run related to the current run folder is retrieved from the
+tracking database with this tuple attributes.
+The retrieved record must be unique in the DB, otherwise it exits
+with error.
+When there is no record in the DB, a new run record is created
+by assigning the main tuple's attributes from the RunParameters.json
```

Comment:
'by assigning the main tuple's attributes from the RunParameters.json' using run attributes from RunParameters.json